opaque_keys/edx/locator.py

```diff
@@ -151,6 +153,10 @@ class CourseLocator(BlockLocatorBase, CourseKey):   # pylint: disable=abstract-m
     __slots__ = KEY_FIELDS
     CHECKED_INIT = False
 
+    # A mapping of parsed key fields to CourseLocator objects, for caching.
+    # Every subclass has its own.
+    __parsed_fields_to_keys = WeakValueDictionary()
```

Comment:
Ugh, you're right. That was me misremembering the feature. That makes me very sad. I'll try a couple of alternatives to see how they perform. Thank you. :)